platform/jvm/Options.kt

```diff
@@ -0,0 +1,224 @@
+            )
+        }
+    }
+
+    data class Launcher(val mainJarPath: String, val mainClass: String, val createShortcut: Boolean, val linux: Linux, val windows: Windows) {
+        companion object {
+            fun of(props: Properties) = Launcher(
+                mainJarPath = props.requireString(MAIN_JAR_PATH),
+                mainClass = props.requireString(MAIN_CLASS),
+                createShortcut = props.getBooleanOrDefault(CREATE_SHORTCUT, defaultValue = false),
+                linux = Linux.of(props),
+                windows = Windows.of(props)
+            )
+        }
+
+        data class Linux(val menuGroup: String?, val appCategory: String?) {
+            companion object {
+                fun of(props: Properties) = Linux(
+                    menuGroup = props.getStringOrNull(LINUX_MENU_GROUP),
+                    appCategory = props.getStringOrNull(LINUX_APP_CATEGORY)
+                )
+            }
+        }
+
+        data class Windows(val menuGroup: String?) {
+            companion object {
+                fun of(props: Properties) = Windows(menuGroup = props.getStringOrNull(WINDOWS_MENU_GROUP))
+            }
+        }
+    }
+
+    data class Output(val archivePath: String) {
+        companion object {
+            fun of(props: Properties) = Output(archivePath = props.requireString(OUTPUT_ARCHIVE_PATH))
+        }
+    }
+
+    data class AppleCodeSigning(
+        val appleID: String, val appleIDPassword: String, val cert: File, val certPassword: String,
+        val macAppID: String, val deepSignJarsRegex: Regex?
+    ) {
+        val signNativeLibsInDeps: Boolean = deepSignJarsRegex != null
+
+        companion object {
+            fun of(commandLineParams: CommandLineParams, props: Properties) = commandLineParams.run {
+                AppleCodeSigning(
+                    appleID = require(APPLE_ID, appleID),
+                    appleIDPassword = require(APPLE_ID_PASSWORD, appleIDPassword),
+                    cert = File(props.requireString(APPLE_CODE_SIGNING_CERT_PATH)),
+                    certPassword = require(APPLE_CODE_SIGNING_CERT_PASSWORD, appleCodeSigningCertPassword),
+                    macAppID = props.requireString(MAC_APP_ID),
+                    deepSignJarsRegex = props.getStringOrNull(APPLE_DEEP_SIGN_JARS_REGEX)?.let { Regex(it) }
+                )
+            }
+
+            private fun <T> require(key: String, value: T?): T {
+                if (value == null || value is String && value.isBlank()) {
+                    throw IllegalStateException("'$key' must be set if '${APPLE_CODE_SIGNING_CERT_PATH}' is set")
+                }
+                return value
+            }
+        }
+
+        override fun toString(): String {
+            return "Options.AppleCodeSigning: macAppID=$macAppID, deepSignJarsRegex=$deepSignJarsRegex; (credentials hidden)"
+        }
+    }
+
+    private object Keys {
```

Comment:
These are all the keys in the `.properties` file that gets parsed into this Options object.